api/v1alpha3/virtualmachine_types.go

```diff
@@ -154,6 +154,14 @@ const (
 	ManagedByExtensionType = "VirtualMachine"
 )
 
+const (
+	// VirtualMachineBackupCondition exposes the status of the latest VirtualMachine Backup, when available.
+	VirtualMachineBackupLatestCondition = "VirtualMachineBackupLatest"
```

Comment:
```suggestion
	VirtualMachineBackupUpToDateCondition = "VirtualMachineBackupUpToDate"
```

Kind of like: https://github.com/kubernetes-sigs/cluster-api/blob/main/controlplane/kubeadm/api/v1beta1/condition_consts.go#L52


And then, the reasons should be distinct as well.  Backup not up-to-date because a restore happened is different from an error comparing the backup version.
Same as how CAPI has different reasons for the uptodate condition:
https://github.com/kubernetes-sigs/cluster-api/blob/main/controlplane/kubeadm/internal/controllers/controller.go#L416

```
	// RollingUpdateInProgressReason (Severity=Warning) documents a KubeadmControlPlane object executing a
	// rolling upgrade for aligning the machines spec to the desired state.
	RollingUpdateInProgressReason = "RollingUpdateInProgress"
```

Reply:
Thanks. Updated and added a new FailedReason for err != nil . PTAL